complete/helpers.c

```diff
@@ -58,7 +58,7 @@ void matches_ensure_morespace(struct CompletionData *cd, int new_size)
 
   new_size = ROUND_UP(new_size + 2, 512);
 
-  mutt_mem_realloc(&cd->match_list, new_size * sizeof(char *));
+  MUTT_MEM_REALLOC(&cd->match_list, new_size, const char *);
```

Comment:
This is one example where I had to add a `const` to exactly match the type of the pointer to which we assign, to avoid a type mismatch.

@gahr